.github/workflows/test.yml

```diff
@@ -46,3 +46,15 @@ jobs:
       uses: codecov/codecov-action@v5
       with:
         name: Python ${{ matrix.python-version }}
+
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - uses: astral-sh/ruff-action@v3
+      with:
+        version: "latest"
+    - name: Ruff lint
+      run: ruff check --fix
+    - name: Ruff format
+      run: ruff format
```

Comment:
I'd recommend adding an action for pre-commit hooks, instead of a ruff-only config.